Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -125,60 +129,80 @@ namespace GradientSignal
                 // imageBoundsY)
                 AZ::Vector3 uvw(static_cast<float>(x), static_cast<float>((imageResolutionY - 1) - y), 0.0f);
 
-                GradientSampleParams sampleParams;
-                sampleParams.m_position = inputBoundsStart + (uvw * pixelToBoundsScale) + scaledTexelOffset;
+                AZ::Vector3 position = inputBoundsStart + (uvw * pixelToBoundsScale) + scaledTexelOffset;
 
                 bool inBounds = true;
                 LmbrCentral::ShapeComponentRequestsBus::EventResult(
-                    inBounds, m_boundsEntityId, &LmbrCentral::ShapeComponentRequestsBus::Events::IsPointInside, sampleParams.m_position);
+                    inBounds, m_boundsEntityId, &LmbrCentral::ShapeComponentRequestsBus::Events::IsPointInside, position);
```

Comment:
So would that mean querying the `GetEncompassingAabb` once before the loop and then just checking if it contains the point? Does `GetEncompassingAabb` return an invalid `Aabb` for more complex shapes?

Reply:
It's a bit of work and would definitely be another PR
- Add a region based GetValues() query to gradient sampler (see TerrainSystem for examples)
- Query the shape type to see if it's an AABB
- If so, just get its bounds directly and construct a region query. 
This would avoid the need for storing positions or indices.

Reply:
Nice OK yes I can make a note to take a look at this in a future task!